File: 프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.py
def solution(elements):
    answer = 0
    s = set()
    elements = elements * 2
    for i in range(1, len(elements) // 2 + 1) :
        for j in range(1, len(elements) // 2 + 1) :
            s.add(sum(elements[i:i + j]))
        
    return len(s)

File: 프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.java
import java.util.*;

class Solution {
    public int solution(int[] elements) {
        int answer = 0;
        HashSet<Integer> set = new HashSet<>();
        for (int length = 1; length <= elements.length; length++) { // 길이
            for (int i = 0; i < elements.length; i++) { // 기준점
                int start = i;
                int end = i + length;
                int temp = 0;
                for (int j = start; j < end; j++) {
                    int index = j % elements.length;
                    temp += elements[index];
                }
                set.add(temp);
            }
        }
        answer = set.size();
        return answer;
    }
}